px-dotnet/Resources/Preapproval.cs

```diff
@@ -1,10 +1,208 @@
-﻿using System;
+﻿using MercadoPago.DataStructures.Preapproval;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+
 namespace MercadoPago.Resources
 {
-    public class Preapproval
+    public class Preapproval : MPBase
     {
-        public Preapproval()
+
+        #region Properties
+
+        [StringLength(256)]
+        private string _payer_email;
+        [StringLength(500)]
+        private string _back_url;
+        private string _id;
+        private string _init_point;
+        private string _sandbox_init_point;
+        [StringLength(256)]
+        private string _external_reference;
+        private string _reason;
+        private AutoRecurring? _auto_recurring;
+
+        #endregion
+
+        #region Accesors
+
+        /// <summary>
+        ///  identifier
+        /// </summary>
+        public string Id
         {
+            get
+            {
+                return _id;
+            }
+
+            set
+            {
+                _id = value;
+            }
         }
+        /// <summary>
+        ///  Buy reason
+        /// </summary>
+        public string Reason
+        {
+            get
+            {
+                return _reason;
+            }
+
+            set
+            {
+                _reason = value;
+            }
+        }
+
+        /// <summary>
+        ///  Payer Email
+        /// </summary>
+        public string Payer_email
```

Comment:
We prefer [PascalCase](https://msdn.microsoft.com/en-us/library/ms229043(v=vs.100).aspx). : PayerEmail